<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Make PICRUSt2 Output Analysis and Visualization Easier â€¢ ggpicrust2</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="bootstrap-toc.css">
<script src="bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="pkgdown.css" rel="stylesheet">
<script src="pkgdown.js"></script><meta property="og:title" content="Make PICRUSt2 Output Analysis and Visualization Easier">
<meta property="og:description" content="Provides a convenient way to analyze and visualize PICRUSt2 output with pre-defined plots and functions. Allows for generating statistical plots about microbiome functional predictions and offers customization options. Features a one-click option for creating publication-level plots, saving time and effort in producing professional-grade figures. Streamlines the PICRUSt2 analysis and visualization process. For more details, see Yang et al. (2023) &lt;doi:10.1093/bioinformatics/btad470&gt;.">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-home">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="index.html">ggpicrust2</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.7.2</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="articles/using_ggpicrust2.html">ggpicrust2 vigenett</a>
    </li>
  </ul>
</li>
<li>
  <a href="news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/cafferychen777/ggpicrust2/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="contents col-md-9">
<div class="section level1">
<div class="page-header"><h1 id="ggpicrust2-vignettes">ggpicrust2 vignettes<a class="anchor" aria-label="anchor" href="#ggpicrust2-vignettes"></a>
</h1></div>
<p><em>ggpicrust2</em> is a comprehensive package designed to provide a seamless and intuitive solution for analyzing and interpreting the results of PICRUSt2 functional prediction. It offers a wide range of features, including pathway name/description annotations, advanced differential abundance (DA) methods, and visualization of DA results.</p>
<p>One of the newest additions to <em>ggpicrust2</em> is the capability to compare the consistency and inconsistency across different DA methods applied to the same dataset. This feature allows users to assess the agreement and discrepancy between various methods when it comes to predicting and sequencing the metagenome of a particular sample. It provides valuable insights into the consistency of results obtained from different approaches and helps users evaluate the robustness of their findings.</p>
<p>By leveraging this functionality, researchers, data scientists, and bioinformaticians can gain a deeper understanding of the underlying biological processes and mechanisms present in their PICRUSt2 output data. This comparison of different methods enables them to make informed decisions and draw reliable conclusions based on the consistency evaluation of macrogenomic predictions or sequencing results for the same sample.</p>
<p>If you are interested in exploring and analyzing your PICRUSt2 output data, <em>ggpicrust2</em> is a powerful tool that provides a comprehensive set of features, including the ability to assess the consistency and evaluate the performance of different methods applied to the same dataset.</p>
<p><a href="https://CRAN.R-project.org/package=ggpicrust2" class="external-link"><img src="https://www.r-pkg.org/badges/version/ggpicrust2" alt="CRAN version"></a> <a href="https://CRAN.R-project.org/package=ggpicrust2" class="external-link"><img src="https://cranlogs.r-pkg.org/badges/grand-total/ggpicrust2" alt="Downloads"></a> <a href="https://opensource.org/license/mit/" class="external-link"><img src="https://img.shields.io/badge/License-MIT-yellow.svg" alt="License: MIT"></a></p>
<div class="section level2">
<h2 id="news">News<a class="anchor" aria-label="anchor" href="#news"></a>
</h2>
<p>Update alert! ðŸš¨ ggpicrust2 Ver1.7.2 is now available. To get the latest version, simply use</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://remotes.r-lib.org/reference/install_github.html" class="external-link">install_github</a></span><span class="op">(</span><span class="st">'cafferychen777/ggpicrust2'</span><span class="op">)</span></span></code></pre></div>
<p>Donâ€™t miss out on the major bug fixes and improvements. ðŸ˜€</p>
<p><strong>Join our official Discord channel for more updates and discussions:</strong> <a href="https://discord.gg/yaR7FEc8" class="external-link">ggpicrust2 Discord</a></p>
</div>
<div class="section level2">
<h2 id="table-of-contents">Table of Contents<a class="anchor" aria-label="anchor" href="#table-of-contents"></a>
</h2>
<ul>
<li><a href="#citation">Citation</a></li>
<li><a href="#installation">Installation</a></li>
<li><a href="#stay-updated">Stay Updated</a></li>
<li><a href="#workflow">Workflow</a></li>
<li><a href="#output">Output</a></li>
<li>
<a href="#function-details">Function Details</a>
<ul>
<li><a href="#ko2kegg_abundance">ko2kegg_abundance()</a></li>
<li><a href="#pathway_daa">pathway_daa()</a></li>
<li><a href="#compare_daa_results">compare_daa_results()</a></li>
<li><a href="#pathway_annotation">pathway_annotation()</a></li>
<li><a href="#pathway_errorbar">pathway_errorbar()</a></li>
<li><a href="#pathway_heatmap">pathway_heatmap()</a></li>
<li><a href="#pathway_pca">pathway_pca()</a></li>
<li><a href="#compare_metagenome_results">compare_metagenome_results()</a></li>
</ul>
</li>
<li><a href="#faq">FAQ</a></li>
<li><a href="#authors-other-projects">Authorâ€™s Other Projects</a></li>
</ul>
</div>
<div class="section level2">
<h2 id="citation">Citation<a class="anchor" aria-label="anchor" href="#citation"></a>
</h2>
<p>If you use <em>ggpicrust2</em> in your research, please cite the following paper:</p>
<p>Chen Yang and others. (2023). ggpicrust2: an R package for PICRUSt2 predicted functional profile analysis and visualization. <em>Bioinformatics</em>, btad470. <a href="https://doi.org/10.1093/bioinformatics/btad470" class="external-link">DOI link</a></p>
<p>BibTeX entry: <a href="https://academic.oup.com/bioinformatics/advance-article/doi/10.1093/bioinformatics/btad470/7234609?utm_source=advanceaccess&amp;utm_campaign=bioinformatics&amp;utm_medium=email&amp;login=false" class="external-link">Download here</a></p>
<p>ResearchGate link: <a href="https://www.researchgate.net/publication/372829051_ggpicrust2_an_R_package_for_PICRUSt2_predicted_functional_profile_analysis_and_visualization" class="external-link">Click here</a></p>
<p>Bioinformatics link: <a href="https://academic.oup.com/bioinformatics/advance-article/doi/10.1093/bioinformatics/btad470/7234609?utm_source=advanceaccess&amp;utm_campaign=bioinformatics&amp;utm_medium=email&amp;login=false" class="external-link">Click here</a></p>
</div>
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<p>You can install the stable version of <em>ggpicrust2</em> from CRAN with:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"ggpicrust2"</span><span class="op">)</span></span></code></pre></div>
<p>To install the latest development version of <em>ggpicrust2</em> from GitHub, you can use:</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Install the devtools package if not already installed</span></span>
<span><span class="co"># install.packages("devtools")</span></span>
<span></span>
<span><span class="co"># Install ggpicrust2 from GitHub</span></span>
<span><span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://remotes.r-lib.org/reference/install_github.html" class="external-link">install_github</a></span><span class="op">(</span><span class="st">"cafferychen777/ggpicrust2"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="dependent-cran-packages">Dependent CRAN Packages<a class="anchor" aria-label="anchor" href="#dependent-cran-packages"></a>
</h2>
<table class="table">
<colgroup>
<col width="14%">
<col width="85%">
</colgroup>
<thead><tr class="header">
<th>Package</th>
<th>Description</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>aplot</td>
<td>Create interactive plots</td>
</tr>
<tr class="even">
<td>dplyr</td>
<td>A fast consistent tool for working with data frame like objects both in memory and out of memory</td>
</tr>
<tr class="odd">
<td>ggplot2</td>
<td>An implementation of the Grammar of Graphics in R</td>
</tr>
<tr class="even">
<td>grid</td>
<td>A rewrite of the graphics layout capabilities of R</td>
</tr>
<tr class="odd">
<td>MicrobiomeStat</td>
<td>Statistical analysis of microbiome data</td>
</tr>
<tr class="even">
<td>readr</td>
<td>Read rectangular data (csv tsv fwf) into R</td>
</tr>
<tr class="odd">
<td>stats</td>
<td>The R Stats Package</td>
</tr>
<tr class="even">
<td>tibble</td>
<td>Simple Data Frames</td>
</tr>
<tr class="odd">
<td>tidyr</td>
<td>Easily tidy data with spread() and gather() functions</td>
</tr>
<tr class="even">
<td>ggprism</td>
<td>Interactive 3D plots with â€˜prismâ€™ graphics</td>
</tr>
<tr class="odd">
<td>cowplot</td>
<td>Streamlined Plot Theme and Plot Annotations for â€˜ggplot2â€™</td>
</tr>
<tr class="even">
<td>ggforce</td>
<td>Easily add secondary axes, zooms, and image overlays to â€˜ggplot2â€™</td>
</tr>
<tr class="odd">
<td>ggplotify</td>
<td>Convert complex plots into â€˜grobâ€™ or â€˜ggplotâ€™ objects</td>
</tr>
<tr class="even">
<td>magrittr</td>
<td>A Forward-Pipe Operator for R</td>
</tr>
<tr class="odd">
<td>utils</td>
<td>The R Utils Package</td>
</tr>
</tbody>
</table>
</div>
<div class="section level2">
<h2 id="dependent-bioconductor-packages">Dependent Bioconductor Packages<a class="anchor" aria-label="anchor" href="#dependent-bioconductor-packages"></a>
</h2>
<table class="table">
<colgroup>
<col width="23%">
<col width="76%">
</colgroup>
<thead><tr class="header">
<th>Package</th>
<th>Description</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>phyloseq</td>
<td>Handling and analysis of high-throughput microbiome census data</td>
</tr>
<tr class="even">
<td>ALDEx2</td>
<td>Differential abundance analysis of taxonomic and functional features</td>
</tr>
<tr class="odd">
<td>SummarizedExperiment</td>
<td>SummarizedExperiment container for storing data and metadata together</td>
</tr>
<tr class="even">
<td>Biobase</td>
<td>Base functions for Bioconductor</td>
</tr>
<tr class="odd">
<td>devtools</td>
<td>Tools to make developing R packages easier</td>
</tr>
<tr class="even">
<td>ComplexHeatmap</td>
<td>Making Complex Heatmaps in R</td>
</tr>
<tr class="odd">
<td>BiocGenerics</td>
<td>S4 generic functions for Bioconductor</td>
</tr>
<tr class="even">
<td>BiocManager</td>
<td>Access the Bioconductor Project Package Repositories</td>
</tr>
<tr class="odd">
<td>metagenomeSeq</td>
<td>Statistical analysis for sparse high-throughput sequencing</td>
</tr>
<tr class="even">
<td>Maaslin2</td>
<td>Tools for microbiome analysis</td>
</tr>
<tr class="odd">
<td>edgeR</td>
<td>Empirical Analysis of Digital Gene Expression Data in R</td>
</tr>
<tr class="even">
<td>lefser</td>
<td>R implementation of the LEfSE method for microbiome biomarker discovery</td>
</tr>
<tr class="odd">
<td>limma</td>
<td>Linear Models for Microarray and RNA-Seq Data</td>
</tr>
<tr class="even">
<td>KEGGREST</td>
<td>R Interface to KEGG REST API</td>
</tr>
<tr class="odd">
<td>DESeq2</td>
<td>Differential gene expression analysis using RNA-seq data</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html" class="external-link">requireNamespace</a></span><span class="op">(</span><span class="st">"BiocManager"</span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">pkgs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"phyloseq"</span>, <span class="st">"ALDEx2"</span>, <span class="st">"SummarizedExperiment"</span>, <span class="st">"Biobase"</span>, <span class="st">"devtools"</span>, </span>
<span>          <span class="st">"ComplexHeatmap"</span>, <span class="st">"BiocGenerics"</span>, <span class="st">"BiocManager"</span>, <span class="st">"metagenomeSeq"</span>, </span>
<span>          <span class="st">"Maaslin2"</span>, <span class="st">"edgeR"</span>, <span class="st">"lefser"</span>, <span class="st">"limma"</span>, <span class="st">"KEGGREST"</span>, <span class="st">"DESeq2"</span><span class="op">)</span></span>
<span></span>
<span><span class="kw">for</span> <span class="op">(</span><span class="va">pkg</span> <span class="kw">in</span> <span class="va">pkgs</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html" class="external-link">requireNamespace</a></span><span class="op">(</span><span class="va">pkg</span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span></span>
<span>    <span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://bioconductor.github.io/BiocManager/reference/install.html" class="external-link">install</a></span><span class="op">(</span><span class="va">pkg</span><span class="op">)</span></span>
<span><span class="op">}</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="stay-updated">Stay Updated<a class="anchor" aria-label="anchor" href="#stay-updated"></a>
</h2>
<p>Stay up to date with the latest <em>ggpicrust2</em> developments by following me on Twitter: <a href="https://twitter.com/CafferyYang" class="external-link"><img src="https://img.shields.io/twitter/follow/CafferyYang?style=social"></a></p>
<p>On my Twitter account, youâ€™ll find regular updates, announcements, and insights related to <em>ggpicrust2</em>. By following me, youâ€™ll ensure that you never miss any important information or new features.</p>
<p>Feel free to join the conversation, ask questions, and engage with other users who are also interested in <em>ggpicrust2</em>. Twitter is a great platform to stay connected and be a part of the community.</p>
<p>Click on the Twitter follow button above or visit <a href="https://twitter.com/CafferyYang" class="external-link uri">https://twitter.com/CafferyYang</a> to follow me now!</p>
<p>Thank you for your interest in <em>ggpicrust2</em>, and I look forward to keeping you informed about all the exciting updates!</p>
</div>
<div class="section level2">
<h2 id="workflow">Workflow<a class="anchor" aria-label="anchor" href="#workflow"></a>
</h2>
<p>The easiest way to analyze the PICRUSt2 output is using ggpicrust2() function. The main pipeline can be run with ggpicrust2() function.</p>
<p>ggpicrust2() integrates ko abundance to kegg pathway abundance conversion, annotation of pathway, differential abundance (DA) analysis, part of DA results visualization. When you have trouble running ggpicrust2(), you can debug it by running a separate function, which will greatly increase the speed of your analysis and visualization.</p>
<p><img src="https://raw.githubusercontent.com/cafferychen777/ggpicrust2_paper/main/paper_figure/Workflow.png"></p>
<div class="section level3">
<h3 id="ggpicrust2">ggpicrust2()<a class="anchor" aria-label="anchor" href="#ggpicrust2"></a>
</h3>
<p>You can download the example dataset from the provided <a href="https://github.com/cafferychen777/ggpicrust2_paper/tree/main/Dataset" class="external-link">Github link</a> and <a href="https://drive.google.com/drive/folders/1on4RKgm9NkaBCykMCCRvVJuEJeNVVqAF?usp=share_link" class="external-link">Google Drive link</a> or use the dataset included in the package.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># If you want to analyze the abundance of KEGG pathways instead of KO within the pathway, please set `ko_to_kegg` to TRUE.</span></span>
<span><span class="co"># KEGG pathways typically have more descriptive explanations.</span></span>
<span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://readr.tidyverse.org" class="external-link">readr</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/cafferychen777/ggpicrust2" class="external-link">ggpicrust2</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://tibble.tidyverse.org/" class="external-link">tibble</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://tidyverse.tidyverse.org" class="external-link">tidyverse</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://csdaw.github.io/ggprism/" class="external-link">ggprism</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://patchwork.data-imaginist.com" class="external-link">patchwork</a></span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Load necessary data: abundance data and metadata</span></span>
<span><span class="va">abundance_file</span> <span class="op">&lt;-</span> <span class="st">"path/to/your/abundance_file.tsv"</span></span>
<span><span class="va">metadata</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://readr.tidyverse.org/reference/read_delim.html" class="external-link">read_delim</a></span><span class="op">(</span></span>
<span>    <span class="st">"path/to/your/metadata.txt"</span>,</span>
<span>    delim <span class="op">=</span> <span class="st">"\t"</span>,</span>
<span>    escape_double <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>    trim_ws <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Run ggpicrust2 with input file path</span></span>
<span><span class="va">results_file_input</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/ggpicrust2.html">ggpicrust2</a></span><span class="op">(</span>file <span class="op">=</span> <span class="va">abundance_file</span>,</span>
<span>                                 metadata <span class="op">=</span> <span class="va">metadata</span>,</span>
<span>                                 group <span class="op">=</span> <span class="st">"your_group_column"</span>, <span class="co"># For example dataset, group = "Environment"</span></span>
<span>                                 pathway <span class="op">=</span> <span class="st">"KO"</span>,</span>
<span>                                 daa_method <span class="op">=</span> <span class="st">"LinDA"</span>,</span>
<span>                                 ko_to_kegg <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                                 order <span class="op">=</span> <span class="st">"pathway_class"</span>,</span>
<span>                                 p_values_bar <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                                 x_lab <span class="op">=</span> <span class="st">"pathway_name"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Run ggpicrust2 with imported data.frame</span></span>
<span><span class="va">abundance_data</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://readr.tidyverse.org/reference/read_delim.html" class="external-link">read_delim</a></span><span class="op">(</span><span class="va">abundance_file</span>, delim <span class="op">=</span> <span class="st">"\t"</span>, col_names <span class="op">=</span> <span class="cn">TRUE</span>, trim_ws <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Run ggpicrust2 with input data</span></span>
<span><span class="va">results_data_input</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/ggpicrust2.html">ggpicrust2</a></span><span class="op">(</span>data <span class="op">=</span> <span class="va">abundance_data</span>,</span>
<span>                                 metadata <span class="op">=</span> <span class="va">metadata</span>,</span>
<span>                                 group <span class="op">=</span> <span class="st">"your_group_column"</span>, <span class="co"># For example dataset, group = "Environment"</span></span>
<span>                                 pathway <span class="op">=</span> <span class="st">"KO"</span>,</span>
<span>                                 daa_method <span class="op">=</span> <span class="st">"LinDA"</span>,</span>
<span>                                 ko_to_kegg <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                                 order <span class="op">=</span> <span class="st">"pathway_class"</span>,</span>
<span>                                 p_values_bar <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                                 x_lab <span class="op">=</span> <span class="st">"pathway_name"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Access the plot and results dataframe for the first DA method</span></span>
<span><span class="va">example_plot</span> <span class="op">&lt;-</span> <span class="va">results_file_input</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">$</span><span class="va">plot</span></span>
<span><span class="va">example_results</span> <span class="op">&lt;-</span> <span class="va">results_file_input</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">$</span><span class="va">results</span></span>
<span></span>
<span><span class="co"># Use the example data in ggpicrust2 package</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">ko_abundance</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">metadata</span><span class="op">)</span></span>
<span><span class="va">results_file_input</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/ggpicrust2.html">ggpicrust2</a></span><span class="op">(</span>data <span class="op">=</span> <span class="va">ko_abundance</span>,</span>
<span>                                 metadata <span class="op">=</span> <span class="va">metadata</span>,</span>
<span>                                 group <span class="op">=</span> <span class="st">"Environment"</span>,</span>
<span>                                 pathway <span class="op">=</span> <span class="st">"KO"</span>,</span>
<span>                                 daa_method <span class="op">=</span> <span class="st">"LinDA"</span>,</span>
<span>                                 ko_to_kegg <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                                 order <span class="op">=</span> <span class="st">"pathway_class"</span>,</span>
<span>                                 p_values_bar <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                                 x_lab <span class="op">=</span> <span class="st">"pathway_name"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Analyze the EC or MetaCyc pathway</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">metacyc_abundance</span><span class="op">)</span></span>
<span><span class="va">results_file_input</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/ggpicrust2.html">ggpicrust2</a></span><span class="op">(</span>data <span class="op">=</span> <span class="va">metacyc_abundance</span>,</span>
<span>                                 metadata <span class="op">=</span> <span class="va">metadata</span>,</span>
<span>                                 group <span class="op">=</span> <span class="st">"Environment"</span>,</span>
<span>                                 pathway <span class="op">=</span> <span class="st">"MetaCyc"</span>,</span>
<span>                                 daa_method <span class="op">=</span> <span class="st">"LinDA"</span>,</span>
<span>                                 ko_to_kegg <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>                                 order <span class="op">=</span> <span class="st">"group"</span>,</span>
<span>                                 p_values_bar <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                                 x_lab <span class="op">=</span> <span class="st">"description"</span><span class="op">)</span></span>
<span><span class="va">results_file_input</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">$</span><span class="va">plot</span></span>
<span><span class="va">results_file_input</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">$</span><span class="va">results</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="if-an-error-occurs-with-ggpicrust2-please-use-the-following-workflow">If an error occurs with ggpicrust2, please use the following workflow.<a class="anchor" aria-label="anchor" href="#if-an-error-occurs-with-ggpicrust2-please-use-the-following-workflow"></a>
</h3>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://readr.tidyverse.org" class="external-link">readr</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/cafferychen777/ggpicrust2" class="external-link">ggpicrust2</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://tibble.tidyverse.org/" class="external-link">tibble</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://tidyverse.tidyverse.org" class="external-link">tidyverse</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://csdaw.github.io/ggprism/" class="external-link">ggprism</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://patchwork.data-imaginist.com" class="external-link">patchwork</a></span><span class="op">)</span></span>
<span></span>
<span><span class="co"># If you want to analyze KEGG pathway abundance instead of KO within the pathway, turn ko_to_kegg to TRUE.</span></span>
<span><span class="co"># KEGG pathways typically have more explainable descriptions.</span></span>
<span></span>
<span><span class="co"># Load metadata as a tibble</span></span>
<span><span class="co"># data(metadata)</span></span>
<span><span class="va">metadata</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://readr.tidyverse.org/reference/read_delim.html" class="external-link">read_delim</a></span><span class="op">(</span><span class="st">"path/to/your/metadata.txt"</span>, delim <span class="op">=</span> <span class="st">"\t"</span>, escape_double <span class="op">=</span> <span class="cn">FALSE</span>, trim_ws <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span> </span>
<span></span>
<span><span class="co"># Load KEGG pathway abundance</span></span>
<span><span class="co"># data(kegg_abundance)</span></span>
<span><span class="va">kegg_abundance</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/ko2kegg_abundance.html">ko2kegg_abundance</a></span><span class="op">(</span><span class="st">"path/to/your/pred_metagenome_unstrat.tsv"</span><span class="op">)</span> </span>
<span></span>
<span><span class="co"># Perform pathway differential abundance analysis (DAA) using ALDEx2 method</span></span>
<span><span class="co"># Please change group to "your_group_column" if you are not using example dataset</span></span>
<span><span class="va">daa_results_df</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/pathway_daa.html">pathway_daa</a></span><span class="op">(</span>abundance <span class="op">=</span> <span class="va">kegg_abundance</span>, metadata <span class="op">=</span> <span class="va">metadata</span>, group <span class="op">=</span> <span class="st">"Environment"</span>, daa_method <span class="op">=</span> <span class="st">"ALDEx2"</span>, select <span class="op">=</span> <span class="cn">NULL</span>, reference <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span> </span>
<span></span>
<span><span class="co"># Filter results for ALDEx2_Welch's t test method</span></span>
<span><span class="co"># Please check the unique(daa_results_df$method) and choose one</span></span>
<span><span class="va">daa_sub_method_results_df</span> <span class="op">&lt;-</span> <span class="va">daa_results_df</span><span class="op">[</span><span class="va">daa_results_df</span><span class="op">$</span><span class="va">method</span> <span class="op">==</span> <span class="st">"ALDEx2_Wilcoxon rank test"</span>, <span class="op">]</span></span>
<span></span>
<span><span class="co"># Annotate pathway results using KO to KEGG conversion</span></span>
<span><span class="va">daa_annotated_sub_method_results_df</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/pathway_annotation.html">pathway_annotation</a></span><span class="op">(</span>pathway <span class="op">=</span> <span class="st">"KO"</span>, daa_results_df <span class="op">=</span> <span class="va">daa_sub_method_results_df</span>, ko_to_kegg <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Generate pathway error bar plot</span></span>
<span><span class="co"># Please change Group to metadata$your_group_column if you are not using example dataset</span></span>
<span><span class="va">p</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/pathway_errorbar.html">pathway_errorbar</a></span><span class="op">(</span>abundance <span class="op">=</span> <span class="va">kegg_abundance</span>, daa_results_df <span class="op">=</span> <span class="va">daa_annotated_sub_method_results_df</span>, Group <span class="op">=</span> <span class="va">metadata</span><span class="op">$</span><span class="va">Environment</span>, p_values_threshold <span class="op">=</span> <span class="fl">0.05</span>, order <span class="op">=</span> <span class="st">"pathway_class"</span>, select <span class="op">=</span> <span class="cn">NULL</span>, ko_to_kegg <span class="op">=</span> <span class="cn">TRUE</span>, p_value_bar <span class="op">=</span> <span class="cn">TRUE</span>, colors <span class="op">=</span> <span class="cn">NULL</span>, x_lab <span class="op">=</span> <span class="st">"pathway_name"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># If you want to analyze EC, MetaCyc, and KO without conversions, turn ko_to_kegg to FALSE.</span></span>
<span></span>
<span><span class="co"># Load metadata as a tibble</span></span>
<span><span class="co"># data(metadata)</span></span>
<span><span class="va">metadata</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://readr.tidyverse.org/reference/read_delim.html" class="external-link">read_delim</a></span><span class="op">(</span><span class="st">"path/to/your/metadata.txt"</span>, delim <span class="op">=</span> <span class="st">"\t"</span>, escape_double <span class="op">=</span> <span class="cn">FALSE</span>, trim_ws <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Load KO abundance as a data.frame</span></span>
<span><span class="co"># data(ko_abundance)</span></span>
<span><span class="va">ko_abundance</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.delim</a></span><span class="op">(</span><span class="st">"path/to/your/pred_metagenome_unstrat.tsv"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Perform pathway DAA using ALDEx2 method</span></span>
<span><span class="co"># Please change column_to_rownames() to the feature column if you are not using example dataset</span></span>
<span><span class="co"># Please change group to "your_group_column" if you are not using example dataset</span></span>
<span><span class="va">daa_results_df</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/pathway_daa.html">pathway_daa</a></span><span class="op">(</span>abundance <span class="op">=</span> <span class="va">ko_abundance</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://tibble.tidyverse.org/reference/rownames.html" class="external-link">column_to_rownames</a></span><span class="op">(</span><span class="st">"#NAME"</span><span class="op">)</span>, metadata <span class="op">=</span> <span class="va">metadata</span>, group <span class="op">=</span> <span class="st">"Environment"</span>, daa_method <span class="op">=</span> <span class="st">"ALDEx2"</span>, select <span class="op">=</span> <span class="cn">NULL</span>, reference <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Filter results for ALDEx2_Kruskal-Wallace test method</span></span>
<span><span class="va">daa_sub_method_results_df</span> <span class="op">&lt;-</span> <span class="va">daa_results_df</span><span class="op">[</span><span class="va">daa_results_df</span><span class="op">$</span><span class="va">method</span> <span class="op">==</span> <span class="st">"ALDEx2_Wilcoxon rank test"</span>, <span class="op">]</span></span>
<span></span>
<span><span class="co"># Annotate pathway results without KO to KEGG conversion</span></span>
<span><span class="va">daa_annotated_sub_method_results_df</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/pathway_annotation.html">pathway_annotation</a></span><span class="op">(</span>pathway <span class="op">=</span> <span class="st">"KO"</span>, daa_results_df <span class="op">=</span> <span class="va">daa_sub_method_results_df</span>, ko_to_kegg <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Generate pathway error bar plot</span></span>
<span><span class="co"># Please change column_to_rownames() to the feature column</span></span>
<span><span class="co"># Please change Group to metadata$your_group_column if you are not using example dataset</span></span>
<span><span class="va">p</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/pathway_errorbar.html">pathway_errorbar</a></span><span class="op">(</span>abundance <span class="op">=</span> <span class="va">ko_abundance</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://tibble.tidyverse.org/reference/rownames.html" class="external-link">column_to_rownames</a></span><span class="op">(</span><span class="st">"#NAME"</span><span class="op">)</span>, daa_results_df <span class="op">=</span> <span class="va">daa_annotated_sub_method_results_df</span>, Group <span class="op">=</span> <span class="va">metadata</span><span class="op">$</span><span class="va">Environment</span>, p_values_threshold <span class="op">=</span> <span class="fl">0.05</span>, order <span class="op">=</span> <span class="st">"group"</span>,</span>
<span>select <span class="op">=</span> <span class="va">daa_annotated_sub_method_results_df</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu">arrange</span><span class="op">(</span><span class="va">p_adjust</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu">slice</span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">20</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">feature</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu">pull</span><span class="op">(</span><span class="op">)</span>, </span>
<span>ko_to_kegg <span class="op">=</span> <span class="cn">FALSE</span>, </span>
<span>p_value_bar <span class="op">=</span> <span class="cn">TRUE</span>, </span>
<span>colors <span class="op">=</span> <span class="cn">NULL</span>, </span>
<span>x_lab <span class="op">=</span> <span class="st">"description"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Workflow for MetaCyc Pathway and EC</span></span>
<span></span>
<span><span class="co"># Load MetaCyc pathway abundance and metadata</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"metacyc_abundance"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"metadata"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Perform pathway DAA using LinDA method</span></span>
<span><span class="co"># Please change column_to_rownames() to the feature column if you are not using example dataset</span></span>
<span><span class="co"># Please change group to "your_group_column" if you are not using example dataset</span></span>
<span><span class="va">metacyc_daa_results_df</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/pathway_daa.html">pathway_daa</a></span><span class="op">(</span>abundance <span class="op">=</span> <span class="va">metacyc_abundance</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://tibble.tidyverse.org/reference/rownames.html" class="external-link">column_to_rownames</a></span><span class="op">(</span><span class="st">"pathway"</span><span class="op">)</span>, metadata <span class="op">=</span> <span class="va">metadata</span>, group <span class="op">=</span> <span class="st">"Environment"</span>, daa_method <span class="op">=</span> <span class="st">"LinDA"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Annotate MetaCyc pathway results without KO to KEGG conversion</span></span>
<span><span class="va">metacyc_daa_annotated_results_df</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/pathway_annotation.html">pathway_annotation</a></span><span class="op">(</span>pathway <span class="op">=</span> <span class="st">"MetaCyc"</span>, daa_results_df <span class="op">=</span> <span class="va">metacyc_daa_results_df</span>, ko_to_kegg <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Generate pathway error bar plot</span></span>
<span><span class="co"># Please change column_to_rownames() to the feature column</span></span>
<span><span class="co"># Please change Group to metadata$your_group_column if you are not using example dataset</span></span>
<span><span class="fu"><a href="reference/pathway_errorbar.html">pathway_errorbar</a></span><span class="op">(</span>abundance <span class="op">=</span> <span class="va">metacyc_abundance</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://tibble.tidyverse.org/reference/rownames.html" class="external-link">column_to_rownames</a></span><span class="op">(</span><span class="st">"pathway"</span><span class="op">)</span>, daa_results_df <span class="op">=</span> <span class="va">metacyc_daa_annotated_results_df</span>, Group <span class="op">=</span> <span class="va">metadata</span><span class="op">$</span><span class="va">Environment</span>, ko_to_kegg <span class="op">=</span> <span class="cn">FALSE</span>, p_values_threshold <span class="op">=</span> <span class="fl">0.05</span>, order <span class="op">=</span> <span class="st">"group"</span>, select <span class="op">=</span> <span class="cn">NULL</span>, p_value_bar <span class="op">=</span> <span class="cn">TRUE</span>, colors <span class="op">=</span> <span class="cn">NULL</span>, x_lab <span class="op">=</span> <span class="st">"description"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Generate pathway heatmap</span></span>
<span><span class="co"># Please change column_to_rownames() to the feature column if you are not using example dataset</span></span>
<span><span class="co"># Please change group to "your_group_column" if you are not using example dataset</span></span>
<span><span class="va">feature_with_p_0.05</span> <span class="op">&lt;-</span> <span class="va">metacyc_daa_results_df</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">p_adjust</span> <span class="op">&lt;</span> <span class="fl">0.05</span><span class="op">)</span></span>
<span><span class="fu"><a href="reference/pathway_heatmap.html">pathway_heatmap</a></span><span class="op">(</span>abundance <span class="op">=</span> <span class="va">metacyc_abundance</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">pathway</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="va">feature_with_p_0.05</span><span class="op">$</span><span class="va">feature</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://tibble.tidyverse.org/reference/rownames.html" class="external-link">column_to_rownames</a></span><span class="op">(</span><span class="st">"pathway"</span><span class="op">)</span>, metadata <span class="op">=</span> <span class="va">metadata</span>, group <span class="op">=</span> <span class="st">"Environment"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Generate pathway PCA plot</span></span>
<span><span class="co"># Please change column_to_rownames() to the feature column if you are not using example dataset</span></span>
<span><span class="co"># Please change group to "your_group_column" if you are not using example dataset</span></span>
<span><span class="fu"><a href="reference/pathway_pca.html">pathway_pca</a></span><span class="op">(</span>abundance <span class="op">=</span> <span class="va">metacyc_abundance</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://tibble.tidyverse.org/reference/rownames.html" class="external-link">column_to_rownames</a></span><span class="op">(</span><span class="st">"pathway"</span><span class="op">)</span>, metadata <span class="op">=</span> <span class="va">metadata</span>, group <span class="op">=</span> <span class="st">"Environment"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Run pathway DAA for multiple methods</span></span>
<span><span class="co"># Please change column_to_rownames() to the feature column if you are not using example dataset</span></span>
<span><span class="co"># Please change group to "your_group_column" if you are not using example dataset</span></span>
<span><span class="va">methods</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"ALDEx2"</span>, <span class="st">"DESeq2"</span>, <span class="st">"edgeR"</span><span class="op">)</span></span>
<span><span class="va">daa_results_list</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="va">methods</span>, <span class="kw">function</span><span class="op">(</span><span class="va">method</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="fu"><a href="reference/pathway_daa.html">pathway_daa</a></span><span class="op">(</span>abundance <span class="op">=</span> <span class="va">metacyc_abundance</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://tibble.tidyverse.org/reference/rownames.html" class="external-link">column_to_rownames</a></span><span class="op">(</span><span class="st">"pathway"</span><span class="op">)</span>, metadata <span class="op">=</span> <span class="va">metadata</span>, group <span class="op">=</span> <span class="st">"Environment"</span>, daa_method <span class="op">=</span> <span class="va">method</span><span class="op">)</span></span>
<span><span class="op">}</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Compare results across different methods</span></span>
<span><span class="va">comparison_results</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/compare_daa_results.html">compare_daa_results</a></span><span class="op">(</span>daa_results_list <span class="op">=</span> <span class="va">daa_results_list</span>, method_names <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"ALDEx2_Welch's t test"</span>, <span class="st">"ALDEx2_Wilcoxon rank test"</span>, <span class="st">"DESeq2"</span>, <span class="st">"edgeR"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="output">Output<a class="anchor" aria-label="anchor" href="#output"></a>
</h2>
<p>The typical output of the ggpicrust2 is like this.</p>
<p><img src="https://cdn.jsdelivr.net/gh/cafferychen777/ggpicrust2_paper@main/paper_figure/figure1.jpg"></p>
</div>
<div class="section level2">
<h2 id="function-details">function details<a class="anchor" aria-label="anchor" href="#function-details"></a>
</h2>
<div class="section level3">
<h3 id="ko2kegg_abundance">ko2kegg_abundance()<a class="anchor" aria-label="anchor" href="#ko2kegg_abundance"></a>
</h3>
<p>KEGG Orthology(KO) is a classification system developed by the Kyoto Encyclopedia of Genes and Genomes (KEGG) data-base(Kanehisa et al., 2022). It uses a hierarchical structure to classify enzymes based on the reactions they catalyze. To better understand pathwaysâ€™ role in different groups and classify the pathways, the KO abundance table needs to be converted to KEGG pathway abundance. But PICRUSt2 removes the function from PICRUSt. ko2kegg_abundance() can help convert the table.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Sample usage of the ko2kegg_abundance function</span></span>
<span><span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://remotes.r-lib.org/reference/install_github.html" class="external-link">install_github</a></span><span class="op">(</span><span class="st">'cafferychen777/ggpicrust2'</span><span class="op">)</span></span>
<span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/cafferychen777/ggpicrust2" class="external-link">ggpicrust2</a></span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Assume that the KO abundance table is stored in a file named "ko_abundance.tsv"</span></span>
<span><span class="va">ko_abundance_file</span> <span class="op">&lt;-</span> <span class="st">"ko_abundance.tsv"</span></span>
<span></span>
<span><span class="co"># Convert KO abundance to KEGG pathway abundance</span></span>
<span><span class="va">kegg_abundance</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/ko2kegg_abundance.html">ko2kegg_abundance</a></span><span class="op">(</span>file <span class="op">=</span> <span class="va">ko_abundance_file</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Alternatively, if the KO abundance data is already loaded as a data frame named "ko_abundance"</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"ko_abundance"</span><span class="op">)</span></span>
<span><span class="va">kegg_abundance</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/ko2kegg_abundance.html">ko2kegg_abundance</a></span><span class="op">(</span>data <span class="op">=</span> <span class="va">ko_abundance</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># The resulting kegg_abundance data frame can now be used for further analysis and visualization.</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="pathway_daa">pathway_daa()<a class="anchor" aria-label="anchor" href="#pathway_daa"></a>
</h3>
<p>Differential abundance(DA) analysis plays a major role in PICRUSt2 downstream analysis. pathway_daa() integrates almost all DA methods applicable to the predicted functional profile which there excludes ANCOM and ANCOMBC. It includes <a href="https://bioconductor.riken.jp/packages/3.9/bioc/html/ALDEx2.html" class="external-link">ALDEx2</a>(Fernandes et al., 2013), <a href="https://bioconductor.org/packages/release/bioc/html/DESeq2.html" class="external-link">DESeq2</a>(Love et al., 2014), <a href="https://www.bioconductor.org/packages/release/bioc/html/Maaslin2.html" class="external-link">Maaslin2</a>(Mallick et al., 2021), <a href="https://genomebiology.biomedcentral.com/articles/10.1186/s13059-022-02655-5" class="external-link">LinDA</a>(Zhou et al., 2022), <a href="https://bioconductor.org/packages/release/bioc/html/edgeR.html" class="external-link">edgeR</a>(Robinson et al., 2010) , <a href="https://ucdavis-bioinformatics-training.github.io/2018-June-RNA-Seq-Workshop/thursday/DE.html" class="external-link">limma voom</a>(Ritchie et al., 2015), <a href="https://www.bioconductor.org/packages/release/bioc/html/metagenomeSeq.html#:~:text=metagenomeSeq%20is%20designed%20to%20address,the%20testing%20of%20feature%20correlations." class="external-link">metagenomeSeq</a>(Paulson et al., 2013), <a href="http://bioconductor.org/packages/release/bioc/html/lefser.html" class="external-link">Lefser</a>(Segata et al., 2011).</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># The abundance table is recommended to be a data.frame rather than a tibble.</span></span>
<span><span class="co"># The abundance table should have feature names or pathway names as row names, and sample names as column names.</span></span>
<span><span class="co"># You can use the output of ko2kegg_abundance</span></span>
<span><span class="va">ko_abundance_file</span> <span class="op">&lt;-</span> <span class="st">"path/to/your/pred_metagenome_unstrat.tsv"</span></span>
<span><span class="va">kegg_abundance</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/ko2kegg_abundance.html">ko2kegg_abundance</a></span><span class="op">(</span><span class="va">ko_abundance_file</span><span class="op">)</span> <span class="co"># Or use data(kegg_abundance)</span></span>
<span></span>
<span><span class="va">metadata</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://readr.tidyverse.org/reference/read_delim.html" class="external-link">read_delim</a></span><span class="op">(</span><span class="st">"path/to/your/metadata.txt"</span>, delim <span class="op">=</span> <span class="st">"\t"</span>, escape_double <span class="op">=</span> <span class="cn">FALSE</span>, trim_ws <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># The default DAA method is "ALDEx2"</span></span>
<span><span class="co"># Please change group to "your_group_column" if you are not using example dataset</span></span>
<span><span class="va">daa_results_df</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/pathway_daa.html">pathway_daa</a></span><span class="op">(</span>abundance <span class="op">=</span> <span class="va">kegg_abundance</span>, metadata <span class="op">=</span> <span class="va">metadata</span>, group <span class="op">=</span> <span class="st">"Environment"</span>, daa_method <span class="op">=</span> <span class="st">"linDA"</span>, select <span class="op">=</span> <span class="cn">NULL</span>, p.adjust <span class="op">=</span> <span class="st">"BH"</span>, reference <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># If you have more than 3 group levels and want to use the LinDA, limma voom, or Maaslin2 methods, you should provide a reference.</span></span>
<span><span class="va">metadata</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://readr.tidyverse.org/reference/read_delim.html" class="external-link">read_delim</a></span><span class="op">(</span><span class="st">"path/to/your/metadata.txt"</span>, delim <span class="op">=</span> <span class="st">"\t"</span>, escape_double <span class="op">=</span> <span class="cn">FALSE</span>, trim_ws <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Please change group to "your_group_column" if you are not using example dataset</span></span>
<span><span class="va">daa_results_df</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/pathway_daa.html">pathway_daa</a></span><span class="op">(</span>abundance <span class="op">=</span> <span class="va">kegg_abundance</span>, metadata <span class="op">=</span> <span class="va">metadata</span>, group <span class="op">=</span> <span class="st">"Group"</span>, daa_method <span class="op">=</span> <span class="st">"LinDA"</span>, select <span class="op">=</span> <span class="cn">NULL</span>, p.adjust <span class="op">=</span> <span class="st">"BH"</span>, reference <span class="op">=</span> <span class="st">"Harvard BRI"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Other example</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"metacyc_abundance"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"metadata"</span><span class="op">)</span></span>
<span><span class="va">metacyc_daa_results_df</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/pathway_daa.html">pathway_daa</a></span><span class="op">(</span>abundance <span class="op">=</span> <span class="va">metacyc_abundance</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://tibble.tidyverse.org/reference/rownames.html" class="external-link">column_to_rownames</a></span><span class="op">(</span><span class="st">"pathway"</span><span class="op">)</span>, metadata <span class="op">=</span> <span class="va">metadata</span>, group <span class="op">=</span> <span class="st">"Environment"</span>, daa_method <span class="op">=</span> <span class="st">"LinDA"</span>, select <span class="op">=</span> <span class="cn">NULL</span>, p.adjust <span class="op">=</span> <span class="st">"BH"</span>, reference <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="compare_daa_results">compare_daa_results()<a class="anchor" aria-label="anchor" href="#compare_daa_results"></a>
</h3>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/cafferychen777/ggpicrust2" class="external-link">ggpicrust2</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://tidyverse.tidyverse.org" class="external-link">tidyverse</a></span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"metacyc_abundance"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"metadata"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Run pathway_daa function for multiple methods</span></span>
<span><span class="co"># Please change column_to_rownames() to the feature column if you are not using example dataset</span></span>
<span><span class="co"># Please change group to "your_group_column" if you are not using example dataset</span></span>
<span><span class="va">methods</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"ALDEx2"</span>, <span class="st">"DESeq2"</span>, <span class="st">"edgeR"</span><span class="op">)</span></span>
<span><span class="va">daa_results_list</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="va">methods</span>, <span class="kw">function</span><span class="op">(</span><span class="va">method</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="fu"><a href="reference/pathway_daa.html">pathway_daa</a></span><span class="op">(</span>abundance <span class="op">=</span> <span class="va">metacyc_abundance</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://tibble.tidyverse.org/reference/rownames.html" class="external-link">column_to_rownames</a></span><span class="op">(</span><span class="st">"pathway"</span><span class="op">)</span>, metadata <span class="op">=</span> <span class="va">metadata</span>, group <span class="op">=</span> <span class="st">"Environment"</span>, daa_method <span class="op">=</span> <span class="va">method</span><span class="op">)</span></span>
<span><span class="op">}</span><span class="op">)</span></span>
<span></span>
<span><span class="va">method_names</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"ALDEx2"</span>,<span class="st">"DESeq2"</span>, <span class="st">"edgeR"</span><span class="op">)</span></span>
<span><span class="co"># Compare results across different methods</span></span>
<span><span class="va">comparison_results</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/compare_daa_results.html">compare_daa_results</a></span><span class="op">(</span>daa_results_list <span class="op">=</span> <span class="va">daa_results_list</span>, method_names <span class="op">=</span> <span class="va">method_names</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="pathway_annotation">pathway_annotation()<a class="anchor" aria-label="anchor" href="#pathway_annotation"></a>
</h3>
<p><strong>If you are in China and you are using kegg pathway annotation, Please make sure your internet can break through the firewall.</strong></p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="co"># Make sure to check if the features in `daa_results_df` correspond to the selected pathway</span></span>
<span></span>
<span><span class="co"># Annotate KEGG Pathway</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"kegg_abundance"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"metadata"</span><span class="op">)</span></span>
<span><span class="co"># Please change group to "your_group_column" if you are not using example dataset</span></span>
<span><span class="va">daa_results_df</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/pathway_daa.html">pathway_daa</a></span><span class="op">(</span>abundance <span class="op">=</span> <span class="va">kegg_abundance</span>, metadata <span class="op">=</span> <span class="va">metadata</span>, group <span class="op">=</span> <span class="st">"Environment"</span>, daa_method <span class="op">=</span> <span class="st">"LinDA"</span><span class="op">)</span></span>
<span><span class="va">daa_annotated_results_df</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/pathway_annotation.html">pathway_annotation</a></span><span class="op">(</span>pathway <span class="op">=</span> <span class="st">"KO"</span>, daa_results_df <span class="op">=</span> <span class="va">daa_results_df</span>, ko_to_kegg <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Annotate KO</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"ko_abundance"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"metadata"</span><span class="op">)</span></span>
<span><span class="co"># Please change column_to_rownames() to the feature column if you are not using example dataset</span></span>
<span><span class="co"># Please change group to "your_group_column" if you are not using example dataset</span></span>
<span><span class="va">daa_results_df</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/pathway_daa.html">pathway_daa</a></span><span class="op">(</span>abundance <span class="op">=</span> <span class="va">ko_abundance</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://tibble.tidyverse.org/reference/rownames.html" class="external-link">column_to_rownames</a></span><span class="op">(</span><span class="st">"#NAME"</span><span class="op">)</span>, metadata <span class="op">=</span> <span class="va">metadata</span>, group <span class="op">=</span> <span class="st">"Environment"</span>, daa_method <span class="op">=</span> <span class="st">"LinDA"</span><span class="op">)</span></span>
<span><span class="va">daa_annotated_results_df</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/pathway_annotation.html">pathway_annotation</a></span><span class="op">(</span>pathway <span class="op">=</span> <span class="st">"KO"</span>, daa_results_df <span class="op">=</span> <span class="va">daa_results_df</span>, ko_to_kegg <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Annotate KEGG</span></span>
<span><span class="co"># daa_annotated_results_df &lt;- pathway_annotation(pathway = "EC", daa_results_df = daa_results_df, ko_to_kegg = FALSE)</span></span>
<span></span>
<span><span class="co"># Annotate MetaCyc Pathway</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"metacyc_abundance"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"metadata"</span><span class="op">)</span></span>
<span><span class="co"># Please change column_to_rownames() to the feature column if you are not using example dataset</span></span>
<span><span class="co"># Please change group to "your_group_column" if you are not using example dataset</span></span>
<span><span class="va">metacyc_daa_results_df</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/pathway_daa.html">pathway_daa</a></span><span class="op">(</span>abundance <span class="op">=</span> <span class="va">metacyc_abundance</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://tibble.tidyverse.org/reference/rownames.html" class="external-link">column_to_rownames</a></span><span class="op">(</span><span class="st">"pathway"</span><span class="op">)</span>, metadata <span class="op">=</span> <span class="va">metadata</span>, group <span class="op">=</span> <span class="st">"Environment"</span>, daa_method <span class="op">=</span> <span class="st">"LinDA"</span><span class="op">)</span></span>
<span><span class="va">metacyc_daa_annotated_results_df</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/pathway_annotation.html">pathway_annotation</a></span><span class="op">(</span>pathway <span class="op">=</span> <span class="st">"MetaCyc"</span>, daa_results_df <span class="op">=</span> <span class="va">metacyc_daa_results_df</span>, ko_to_kegg <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="pathway_errorbar">pathway_errorbar()<a class="anchor" aria-label="anchor" href="#pathway_errorbar"></a>
</h3>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"ko_abundance"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"metadata"</span><span class="op">)</span></span>
<span><span class="va">kegg_abundance</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/ko2kegg_abundance.html">ko2kegg_abundance</a></span><span class="op">(</span>data <span class="op">=</span> <span class="va">ko_abundance</span><span class="op">)</span> <span class="co"># Or use data(kegg_abundance)</span></span>
<span><span class="co"># Please change group to "your_group_column" if you are not using example dataset</span></span>
<span><span class="va">daa_results_df</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/pathway_daa.html">pathway_daa</a></span><span class="op">(</span><span class="va">kegg_abundance</span>, metadata <span class="op">=</span> <span class="va">metadata</span>, group <span class="op">=</span> <span class="st">"Environment"</span>, daa_method <span class="op">=</span> <span class="st">"LinDA"</span><span class="op">)</span></span>
<span><span class="va">daa_annotated_results_df</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/pathway_annotation.html">pathway_annotation</a></span><span class="op">(</span>pathway <span class="op">=</span> <span class="st">"KO"</span>, daa_results_df <span class="op">=</span> <span class="va">daa_results_df</span>, ko_to_kegg <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="co"># Please change Group to metadata$your_group_column if you are not using example dataset</span></span>
<span><span class="va">p</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/pathway_errorbar.html">pathway_errorbar</a></span><span class="op">(</span>abundance <span class="op">=</span> <span class="va">kegg_abundance</span>,</span>
<span>           daa_results_df <span class="op">=</span> <span class="va">daa_annotated_results_df</span>,</span>
<span>           Group <span class="op">=</span> <span class="va">metadata</span><span class="op">$</span><span class="va">Environment</span>,</span>
<span>           ko_to_kegg <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>           p_values_threshold <span class="op">=</span> <span class="fl">0.05</span>,</span>
<span>           order <span class="op">=</span> <span class="st">"pathway_class"</span>,</span>
<span>           select <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>           p_value_bar <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>           colors <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>           x_lab <span class="op">=</span> <span class="st">"pathway_name"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># If you want to analysis the EC. MetaCyc. KO without conversions. </span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"metacyc_abundance"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"metadata"</span><span class="op">)</span></span>
<span><span class="va">metacyc_daa_results_df</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/pathway_daa.html">pathway_daa</a></span><span class="op">(</span>abundance <span class="op">=</span> <span class="va">metacyc_abundance</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://tibble.tidyverse.org/reference/rownames.html" class="external-link">column_to_rownames</a></span><span class="op">(</span><span class="st">"pathway"</span><span class="op">)</span>, metadata <span class="op">=</span> <span class="va">metadata</span>, group <span class="op">=</span> <span class="st">"Environment"</span>, daa_method <span class="op">=</span> <span class="st">"LinDA"</span><span class="op">)</span></span>
<span><span class="va">metacyc_daa_annotated_results_df</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/pathway_annotation.html">pathway_annotation</a></span><span class="op">(</span>pathway <span class="op">=</span> <span class="st">"MetaCyc"</span>, daa_results_df <span class="op">=</span> <span class="va">metacyc_daa_results_df</span>, ko_to_kegg <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="va">p</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/pathway_errorbar.html">pathway_errorbar</a></span><span class="op">(</span>abundance <span class="op">=</span> <span class="va">metacyc_abundance</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://tibble.tidyverse.org/reference/rownames.html" class="external-link">column_to_rownames</a></span><span class="op">(</span><span class="st">"pathway"</span><span class="op">)</span>,</span>
<span>           daa_results_df <span class="op">=</span> <span class="va">metacyc_daa_annotated_results_df</span>,</span>
<span>           Group <span class="op">=</span> <span class="va">metadata</span><span class="op">$</span><span class="va">Environment</span>,</span>
<span>           ko_to_kegg <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>           p_values_threshold <span class="op">=</span> <span class="fl">0.05</span>,</span>
<span>           order <span class="op">=</span> <span class="st">"group"</span>,</span>
<span>           select <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>           p_value_bar <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>           colors <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>           x_lab <span class="op">=</span> <span class="st">"description"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="pathway_heatmap">pathway_heatmap()<a class="anchor" aria-label="anchor" href="#pathway_heatmap"></a>
</h3>
<p>In this section, we will demonstrate how to create a pathway heatmap using the <code>pathway_heatmap</code> function in the ggpicrust2 package. This function visualizes the relative abundance of pathways in different samples.</p>
<p>Use the fake dataset</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Create example functional pathway abundance data</span></span>
<span><span class="va">abundance_example</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">matrix</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/stats/Normal.html" class="external-link">rnorm</a></span><span class="op">(</span><span class="fl">30</span><span class="op">)</span>, nrow <span class="op">=</span> <span class="fl">3</span>, ncol <span class="op">=</span> <span class="fl">10</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">abundance_example</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"Sample"</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">10</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">abundance_example</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"PathwayA"</span>, <span class="st">"PathwayB"</span>, <span class="st">"PathwayC"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Create example metadata</span></span>
<span><span class="co"># Please change your sample id's column name to sample_name</span></span>
<span><span class="va">metadata_example</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>sample_name <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">abundance_example</span><span class="op">)</span>,</span>
<span>                               group <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">factor</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Control"</span>, <span class="st">"Treatment"</span><span class="op">)</span>, each <span class="op">=</span> <span class="fl">5</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Create a heatmap</span></span>
<span><span class="fu"><a href="reference/pathway_heatmap.html">pathway_heatmap</a></span><span class="op">(</span><span class="va">abundance_example</span>, <span class="va">metadata_example</span>, <span class="st">"group"</span><span class="op">)</span></span></code></pre></div>
<p>Use the real dataset</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://tidyverse.tidyverse.org" class="external-link">tidyverse</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/teunbrand/ggh4x" class="external-link">ggh4x</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/cafferychen777/ggpicrust2" class="external-link">ggpicrust2</a></span><span class="op">)</span></span>
<span><span class="co"># Load the data</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"metacyc_abundance"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Load the metadata</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"metadata"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Perform differential abundance analysis</span></span>
<span><span class="va">metacyc_daa_results_df</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/pathway_daa.html">pathway_daa</a></span><span class="op">(</span></span>
<span>  abundance <span class="op">=</span> <span class="va">metacyc_abundance</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://tibble.tidyverse.org/reference/rownames.html" class="external-link">column_to_rownames</a></span><span class="op">(</span><span class="st">"pathway"</span><span class="op">)</span>,</span>
<span>  metadata <span class="op">=</span> <span class="va">metadata</span>,</span>
<span>  group <span class="op">=</span> <span class="st">"Environment"</span>,</span>
<span>  daa_method <span class="op">=</span> <span class="st">"LinDA"</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Annotate the results</span></span>
<span><span class="va">annotated_metacyc_daa_results_df</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/pathway_annotation.html">pathway_annotation</a></span><span class="op">(</span></span>
<span>  pathway <span class="op">=</span> <span class="st">"MetaCyc"</span>,</span>
<span>  daa_results_df <span class="op">=</span> <span class="va">metacyc_daa_results_df</span>,</span>
<span>  ko_to_kegg <span class="op">=</span> <span class="cn">FALSE</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Filter features with p &lt; 0.05</span></span>
<span><span class="va">feature_with_p_0.05</span> <span class="op">&lt;-</span> <span class="va">metacyc_daa_results_df</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">p_adjust</span> <span class="op">&lt;</span> <span class="fl">0.05</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Create the heatmap</span></span>
<span><span class="fu"><a href="reference/pathway_heatmap.html">pathway_heatmap</a></span><span class="op">(</span></span>
<span>  abundance <span class="op">=</span> <span class="va">metacyc_abundance</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>    <span class="fu">right_join</span><span class="op">(</span></span>
<span>      <span class="va">annotated_metacyc_daa_results_df</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu">select</span><span class="op">(</span><span class="fu">all_of</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"feature"</span>,<span class="st">"description"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>,</span>
<span>      by <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"pathway"</span> <span class="op">=</span> <span class="st">"feature"</span><span class="op">)</span></span>
<span>    <span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>    <span class="fu"><a href="https://rdrr.io/r/stats/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">pathway</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="va">feature_with_p_0.05</span><span class="op">$</span><span class="va">feature</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>    <span class="fu">select</span><span class="op">(</span><span class="op">-</span><span class="st">"pathway"</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>    <span class="fu"><a href="https://tibble.tidyverse.org/reference/rownames.html" class="external-link">column_to_rownames</a></span><span class="op">(</span><span class="st">"description"</span><span class="op">)</span>,</span>
<span>  metadata <span class="op">=</span> <span class="va">metadata</span>, </span>
<span>  group <span class="op">=</span> <span class="st">"Environment"</span></span>
<span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="pathway_pca">pathway_pca()<a class="anchor" aria-label="anchor" href="#pathway_pca"></a>
</h3>
<p>In this section, we will demonstrate how to perform Principal Component Analysis (PCA) on functional pathway abundance data and create visualizations of the PCA results using the <code>pathway_pca</code> function in the ggpicrust2 package.</p>
<p>Use the fake dataset</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Create example functional pathway abundance data</span></span>
<span><span class="va">abundance_example</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">matrix</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/stats/Normal.html" class="external-link">rnorm</a></span><span class="op">(</span><span class="fl">30</span><span class="op">)</span>, nrow <span class="op">=</span> <span class="fl">3</span>, ncol <span class="op">=</span> <span class="fl">10</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">abundance_example</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"Sample"</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">10</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">kegg_abundance_example</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"PathwayA"</span>, <span class="st">"PathwayB"</span>, <span class="st">"PathwayC"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Create example metadata</span></span>
<span><span class="va">metadata_example</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>sample_name <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">kegg_abundance_example</span><span class="op">)</span>,</span>
<span>                                group <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">factor</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Control"</span>, <span class="st">"Treatment"</span><span class="op">)</span>, each <span class="op">=</span> <span class="fl">5</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co"># Perform PCA and create visualizations</span></span>
<span><span class="fu"><a href="reference/pathway_pca.html">pathway_pca</a></span><span class="op">(</span>abundance <span class="op">=</span> <span class="va">abundance_example</span>, metadata <span class="op">=</span> <span class="va">metadata_example</span>, <span class="st">"group"</span><span class="op">)</span></span></code></pre></div>
<p>Use the real dataset</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Create example functional pathway abundance data</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"metacyc_abundance"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"metadata"</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="reference/pathway_pca.html">pathway_pca</a></span><span class="op">(</span>abundance <span class="op">=</span> <span class="va">metacyc_abundance</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://tibble.tidyverse.org/reference/rownames.html" class="external-link">column_to_rownames</a></span><span class="op">(</span><span class="st">"pathway"</span><span class="op">)</span>, metadata <span class="op">=</span> <span class="va">metadata</span>, group <span class="op">=</span> <span class="st">"Environment"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="compare_metagenome_results">compare_metagenome_results()<a class="anchor" aria-label="anchor" href="#compare_metagenome_results"></a>
</h3>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/jokergoo/ComplexHeatmap" class="external-link">ComplexHeatmap</a></span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">123</span><span class="op">)</span></span>
<span><span class="co"># First metagenome</span></span>
<span><span class="va">metagenome1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/MathFun.html" class="external-link">abs</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">matrix</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/stats/Normal.html" class="external-link">rnorm</a></span><span class="op">(</span><span class="fl">1000</span><span class="op">)</span>, nrow <span class="op">=</span> <span class="fl">100</span>, ncol <span class="op">=</span> <span class="fl">10</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">metagenome1</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"KO"</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">100</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">metagenome1</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"sample"</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">10</span><span class="op">)</span></span>
<span><span class="co"># Second metagenome</span></span>
<span><span class="va">metagenome2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/MathFun.html" class="external-link">abs</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">matrix</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/stats/Normal.html" class="external-link">rnorm</a></span><span class="op">(</span><span class="fl">1000</span><span class="op">)</span>, nrow <span class="op">=</span> <span class="fl">100</span>, ncol <span class="op">=</span> <span class="fl">10</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">metagenome2</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"KO"</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">100</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">metagenome2</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"sample"</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">10</span><span class="op">)</span></span>
<span><span class="co"># Put the metagenomes into a list</span></span>
<span><span class="va">metagenomes</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="va">metagenome1</span>, <span class="va">metagenome2</span><span class="op">)</span></span>
<span><span class="co"># Define names</span></span>
<span><span class="va">names</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"metagenome1"</span>, <span class="st">"metagenome2"</span><span class="op">)</span></span>
<span><span class="co"># Call the function</span></span>
<span><span class="va">results</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/compare_metagenome_results.html">compare_metagenome_results</a></span><span class="op">(</span><span class="va">metagenomes</span>, <span class="va">names</span><span class="op">)</span></span>
<span><span class="co"># Print the correlation matrix</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">results</span><span class="op">$</span><span class="va">correlation</span><span class="op">$</span><span class="va">cor_matrix</span><span class="op">)</span></span>
<span><span class="co"># Print the p-value matrix</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">results</span><span class="op">$</span><span class="va">correlation</span><span class="op">$</span><span class="va">p_matrix</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="share">Share<a class="anchor" aria-label="anchor" href="#share"></a>
</h2>
<p><a href="https://twitter.com/intent/tweet?url=https%3A%2F%2Fgithub.com%2Fcafferychen777%2Fggpicrust2&amp;text=Check%20out%20this%20awesome%20package%20on%20GitHub%21" class="external-link"><img src="https://img.shields.io/twitter/url?url=https%3A%2F%2Fgithub.com%2Fcafferychen777%2Fggpicrust2&amp;style=social" alt="Twitter"></a></p>
<p><a href="https://www.facebook.com/sharer/sharer.php?u=https%3A%2F%2Fgithub.com%2Fcafferychen777%2Fggpicrust2&amp;quote=Check%20out%20this%20awesome%20package%20on%20GitHub%21" class="external-link"><img src="https://img.shields.io/badge/Share_on-Facebook-1877F2?logo=facebook&amp;style=social" alt="Facebook"></a></p>
<p><a href="https://www.linkedin.com/shareArticle?mini=true&amp;url=https%3A%2F%2Fgithub.com%2Fcafferychen777%2Fggpicrust2&amp;title=Check%20out%20this%20awesome%20package%20on%20GitHub%21" class="external-link"><img src="https://img.shields.io/badge/Share_on-LinkedIn-0077B5?logo=linkedin&amp;style=social" alt="LinkedIn"></a></p>
<p><a href="https://www.reddit.com/submit?url=https%3A%2F%2Fgithub.com%2Fcafferychen777%2Fggpicrust2&amp;title=Check%20out%20this%20awesome%20package%20on%20GitHub%21" class="external-link"><img src="https://img.shields.io/badge/Share_on-Reddit-FF4500?logo=reddit&amp;style=social" alt="Reddit"></a></p>
</div>
<div class="section level2">
<h2 id="faq">FAQ<a class="anchor" aria-label="anchor" href="#faq"></a>
</h2>
<div class="section level3">
<h3 id="issue-1-pathway_errorbar-error">Issue 1: pathway_errorbar error<a class="anchor" aria-label="anchor" href="#issue-1-pathway_errorbar-error"></a>
</h3>
<p>When using <code>pathway_errorbar</code> with the following parameters:</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="reference/pathway_errorbar.html">pathway_errorbar</a></span><span class="op">(</span>abundance <span class="op">=</span> <span class="va">abundance</span>,</span>
<span>                 daa_results_df <span class="op">=</span> <span class="va">daa_results_df</span>,</span>
<span>                 Group <span class="op">=</span> <span class="va">metadata</span><span class="op">$</span><span class="va">Environment</span>,</span>
<span>                 ko_to_kegg <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                 p_values_threshold <span class="op">=</span> <span class="fl">0.05</span>,</span>
<span>                 order <span class="op">=</span> <span class="st">"pathway_class"</span>,</span>
<span>                 select <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>                 p_value_bar <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                 colors <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>                 x_lab <span class="op">=</span> <span class="st">"pathway_name"</span><span class="op">)</span></span></code></pre></div>
<p>You may encounter an error:</p>
<div class="sourceCode" id="cb18"><pre class="sourceCode R"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" tabindex="-1"></a>Error <span class="cf">in</span> <span class="st">`</span><span class="at">ggplot_add()</span><span class="st">`</span><span class="sc">:</span></span>
<span id="cb18-2"><a href="#cb18-2" tabindex="-1"></a><span class="sc">!</span> Can<span class="st">'t add `e2` to a &lt;ggplot&gt; object.</span></span>
<span id="cb18-3"><a href="#cb18-3" tabindex="-1"></a><span class="st">Run `rlang::last_trace()` to see where the error occurred.</span></span></code></pre></div>
<p>Make sure you have the <code>patchwork</code> package loaded:</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://patchwork.data-imaginist.com" class="external-link">patchwork</a></span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="issue-2-guide_trainprism_offset_minor-error">Issue 2: guide_train.prism_offset_minor error<a class="anchor" aria-label="anchor" href="#issue-2-guide_trainprism_offset_minor-error"></a>
</h3>
<p>You may encounter an error with <code>guide_train.prism_offset_minor</code>:</p>
<div class="sourceCode" id="cb20"><pre class="sourceCode R"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" tabindex="-1"></a>Error <span class="cf">in</span> <span class="fu">guide_train.prism_offset_minor</span>(guide, panel_params[[aesthetic]]) <span class="sc">:</span> </span>
<span id="cb20-2"><a href="#cb20-2" tabindex="-1"></a>  No minor breaks exist, guide_prism_offset_minor needs minor breaks to work</span>
<span id="cb20-3"><a href="#cb20-3" tabindex="-1"></a></span>
<span id="cb20-4"><a href="#cb20-4" tabindex="-1"></a>Error <span class="cf">in</span> <span class="fu">get</span>(<span class="fu">as.character</span>(FUN)ï¼Œ<span class="at">mode =</span> <span class="st">"function"</span>object <span class="at">envir =</span> envir)</span>
<span id="cb20-5"><a href="#cb20-5" tabindex="-1"></a>guide_prism_offset_minor<span class="st">' of mode'</span>function<span class="st">' was not found</span></span></code></pre></div>
<p>Ensure that the <code>ggprism</code> package is loaded:</p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://csdaw.github.io/ggprism/" class="external-link">ggprism</a></span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="issue-3-ssl-certificate-problem">Issue 3: SSL certificate problem<a class="anchor" aria-label="anchor" href="#issue-3-ssl-certificate-problem"></a>
</h3>
<p>When encountering the following error:</p>
<div class="sourceCode" id="cb22"><pre class="sourceCode R"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" tabindex="-1"></a>SSL peer certificate or SSH remote key was not OK<span class="sc">:</span> [rest.kegg.jp] SSL certificate problem<span class="sc">:</span> certificate has expired</span></code></pre></div>
<p>If you are in China, make sure your computer network can bypass the firewall.</p>
</div>
<div class="section level3">
<h3 id="issue-4-bad-request-http-400">Issue 4: Bad Request (HTTP 400)<a class="anchor" aria-label="anchor" href="#issue-4-bad-request-http-400"></a>
</h3>
<p>When encountering the following error:</p>
<div class="sourceCode" id="cb23"><pre class="sourceCode R"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1" tabindex="-1"></a>Error <span class="cf">in</span> <span class="fu">.getUrl</span>(url, .flatFileParser) <span class="sc">:</span> Bad <span class="fu">Request</span> (HTTP <span class="dv">400</span>).</span></code></pre></div>
<p>Please restart R session.</p>
</div>
<div class="section level3">
<h3 id="issue-5-error-in-gridcallc_textbounds-asgraphicsannotxlabelxx-xy-">Issue 5: Error in grid.Call(C_textBounds, as.graphicsAnnot(xlabel),x<span class="math inline"><em>x</em>,â€†<em>x</em></span>y, :<a class="anchor" aria-label="anchor" href="#issue-5-error-in-gridcallc_textbounds-asgraphicsannotxlabelxx-xy-"></a>
</h3>
<p>When encountering the following error:</p>
<div class="sourceCode" id="cb24"><pre class="sourceCode R"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1" tabindex="-1"></a>Error <span class="cf">in</span> <span class="fu">grid.Call</span>(C_textBounds, <span class="fu">as.graphicsAnnot</span>(xlabel),x<span class="sc">$</span>x, x<span class="sc">$</span>y, <span class="sc">:</span></span></code></pre></div>
<p>Please having some required fonts installed. You can refer to this <a href="https://stackoverflow.com/questions/71362738/r-error-in-grid-callc-textbounds-as-graphicsannotxlabel-xx-xy-polygo" class="external-link">thread</a>.</p>
</div>
<div class="section level3">
<h3 id="issue-6-visualization-becomes-cluttered-when-there-are-more-than-30-features-of-statistical-significance">Issue 6: Visualization becomes cluttered when there are more than 30 features of statistical significance.<a class="anchor" aria-label="anchor" href="#issue-6-visualization-becomes-cluttered-when-there-are-more-than-30-features-of-statistical-significance"></a>
</h3>
<p>When faced with this issue, consider the following solutions:</p>
<p><strong>Solution 1: Utilize the â€˜selectâ€™ parameter</strong></p>
<p>The â€˜selectâ€™ parameter allows you to specify which features you wish to visualize. Hereâ€™s an example of how you can apply this in your code:</p>
<div class="sourceCode" id="cb25"><pre class="sourceCode R"><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1" tabindex="-1"></a>ggpicrust2<span class="sc">::</span><span class="fu">pathway_errorbar</span>(</span>
<span id="cb25-2"><a href="#cb25-2" tabindex="-1"></a>  <span class="at">abundance =</span> kegg_abundance,</span>
<span id="cb25-3"><a href="#cb25-3" tabindex="-1"></a>  <span class="at">daa_results_df =</span> daa_results_df_annotated,</span>
<span id="cb25-4"><a href="#cb25-4" tabindex="-1"></a>  <span class="at">Group =</span> metadata<span class="sc">$</span>Day,</span>
<span id="cb25-5"><a href="#cb25-5" tabindex="-1"></a>  <span class="at">p_values_threshold =</span> <span class="fl">0.05</span>,</span>
<span id="cb25-6"><a href="#cb25-6" tabindex="-1"></a>  <span class="at">order =</span> <span class="st">"pathway_class"</span>,</span>
<span id="cb25-7"><a href="#cb25-7" tabindex="-1"></a>  <span class="at">select =</span> <span class="fu">c</span>(<span class="st">"ko05340"</span>, <span class="st">"ko00564"</span>, <span class="st">"ko00680"</span>, <span class="st">"ko00562"</span>, <span class="st">"ko03030"</span>, <span class="st">"ko00561"</span>, <span class="st">"ko00440"</span>, <span class="st">"ko00250"</span>, <span class="st">"ko00740"</span>, <span class="st">"ko04940"</span>, <span class="st">"ko00010"</span>, <span class="st">"ko00195"</span>, <span class="st">"ko00760"</span>, <span class="st">"ko00920"</span>, <span class="st">"ko00311"</span>, <span class="st">"ko00310"</span>, <span class="st">"ko04146"</span>, <span class="st">"ko00600"</span>, <span class="st">"ko04141"</span>, <span class="st">"ko04142"</span>, <span class="st">"ko00604"</span>, <span class="st">"ko04260"</span>, <span class="st">"ko00909"</span>, <span class="st">"ko04973"</span>, <span class="st">"ko00510"</span>, <span class="st">"ko04974"</span>),</span>
<span id="cb25-8"><a href="#cb25-8" tabindex="-1"></a>  <span class="at">ko_to_kegg =</span> <span class="cn">TRUE</span>,</span>
<span id="cb25-9"><a href="#cb25-9" tabindex="-1"></a>  <span class="at">p_value_bar =</span> <span class="cn">FALSE</span>,</span>
<span id="cb25-10"><a href="#cb25-10" tabindex="-1"></a>  <span class="at">colors =</span> <span class="cn">NULL</span>,</span>
<span id="cb25-11"><a href="#cb25-11" tabindex="-1"></a>  <span class="at">x_lab =</span> <span class="st">"pathway_name"</span></span>
<span id="cb25-12"><a href="#cb25-12" tabindex="-1"></a>)</span></code></pre></div>
<p><strong>Solution 2: Limit to the Top 20 features</strong></p>
<p>If there are too many significant features to visualize effectively, you might consider limiting your visualization to the top 20 features with the smallest adjusted p-values:</p>
<div class="sourceCode" id="cb26"><pre class="sourceCode R"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1" tabindex="-1"></a>daa_results_df_annotated <span class="ot">&lt;-</span> daa_results_df_annotated[<span class="sc">!</span><span class="fu">is.na</span>(daa_results_df_annotated<span class="sc">$</span>pathway_name),]</span>
<span id="cb26-2"><a href="#cb26-2" tabindex="-1"></a></span>
<span id="cb26-3"><a href="#cb26-3" tabindex="-1"></a>daa_results_df_annotated<span class="sc">$</span>p_adjust <span class="ot">&lt;-</span> <span class="fu">round</span>(daa_results_df_annotated<span class="sc">$</span>p_adjust,<span class="dv">5</span>)</span>
<span id="cb26-4"><a href="#cb26-4" tabindex="-1"></a></span>
<span id="cb26-5"><a href="#cb26-5" tabindex="-1"></a>low_p_feature <span class="ot">&lt;-</span> daa_results_df_annotated[<span class="fu">order</span>(daa_results_df_annotated<span class="sc">$</span>p_adjust), ]<span class="sc">$</span>feature[<span class="dv">1</span><span class="sc">:</span><span class="dv">20</span>]</span>
<span id="cb26-6"><a href="#cb26-6" tabindex="-1"></a></span>
<span id="cb26-7"><a href="#cb26-7" tabindex="-1"></a></span>
<span id="cb26-8"><a href="#cb26-8" tabindex="-1"></a>p <span class="ot">&lt;-</span> ggpicrust2<span class="sc">::</span><span class="fu">pathway_errorbar</span>(</span>
<span id="cb26-9"><a href="#cb26-9" tabindex="-1"></a>  <span class="at">abundance =</span> kegg_abundance,</span>
<span id="cb26-10"><a href="#cb26-10" tabindex="-1"></a>  <span class="at">daa_results_df =</span> daa_results_df_annotated,</span>
<span id="cb26-11"><a href="#cb26-11" tabindex="-1"></a>  <span class="at">Group =</span> metadata<span class="sc">$</span>Day,</span>
<span id="cb26-12"><a href="#cb26-12" tabindex="-1"></a>  <span class="at">p_values_threshold =</span> <span class="fl">0.05</span>,</span>
<span id="cb26-13"><a href="#cb26-13" tabindex="-1"></a>  <span class="at">order =</span> <span class="st">"pathway_class"</span>,</span>
<span id="cb26-14"><a href="#cb26-14" tabindex="-1"></a>  <span class="at">select =</span> low_p_feature,</span>
<span id="cb26-15"><a href="#cb26-15" tabindex="-1"></a>  <span class="at">ko_to_kegg =</span> <span class="cn">TRUE</span>,</span>
<span id="cb26-16"><a href="#cb26-16" tabindex="-1"></a>  <span class="at">p_value_bar =</span> <span class="cn">FALSE</span>,</span>
<span id="cb26-17"><a href="#cb26-17" tabindex="-1"></a>  <span class="at">colors =</span> <span class="cn">NULL</span>,</span>
<span id="cb26-18"><a href="#cb26-18" tabindex="-1"></a>  <span class="at">x_lab =</span> <span class="st">"pathway_name"</span></span>
<span id="cb26-19"><a href="#cb26-19" tabindex="-1"></a>)</span></code></pre></div>
</div>
<div class="section level3">
<h3 id="issue-7-there-are-no-statistically-significant-biomarkers">Issue 7: There are no statistically significant biomarkers<a class="anchor" aria-label="anchor" href="#issue-7-there-are-no-statistically-significant-biomarkers"></a>
</h3>
<p>If you are not finding any statistically significant biomarkers in your analysis, there could be several reasons for this:</p>
<ol style="list-style-type: decimal">
<li><p><strong>The true difference between your groups is small or non-existent.</strong> If the microbial communities or pathways youâ€™re comparing are truly similar, then itâ€™s correct and expected that you wonâ€™t find significant differences.</p></li>
<li><p><strong>Your sample size might be too small to detect the differences.</strong> Statistical power, the ability to detect differences if they exist, increases with sample size.</p></li>
<li><p><strong>The variation within your groups might be too large.</strong> If thereâ€™s a lot of variation in microbial communities within a single group, it can be hard to detect differences between groups.</p></li>
</ol>
<p>Here are a few suggestions:</p>
<ol style="list-style-type: decimal">
<li><p><strong>Increase your sample size</strong>: If possible, adding more samples to your analysis can increase your statistical power, making it easier to detect significant differences.</p></li>
<li><p><strong>Decrease intra-group variation</strong>: If thereâ€™s a lot of variation within your groups, consider whether there are outliers or subgroups that are driving this variation. You might need to clean your data, or to stratify your analysis to account for these subgroups.</p></li>
<li><p><strong>Change your statistical method or adjust parameters</strong>: Depending on the nature of your data and your specific question, different statistical methods might be more or less powerful. If youâ€™re currently using a parametric test, consider using a non-parametric test, or vice versa. Also, consider whether adjusting the parameters of your current test might help.</p></li>
</ol>
<p>Remember, not finding significant results is also a result and can be informative, as it might indicate that there are no substantial differences between the groups youâ€™re studying. Itâ€™s important to interpret your results in the context of your specific study and not to force statistical significance where there isnâ€™t any.</p>
<p>With these strategies, you should be able to create a more readable and informative visualization, even when dealing with a large number of significant features.</p>
</div>
</div>
<div class="section level2">
<h2 id="authors-other-projects">Authorâ€™s Other Projects<a class="anchor" aria-label="anchor" href="#authors-other-projects"></a>
</h2>
<ol style="list-style-type: decimal">
<li>
<a href="https://a95dps-caffery-chen.shinyapps.io/MicrobiomeGallery/" class="external-link">MicrobiomeGallery</a>, a <strong>web-based platform for sharing microbiome data visualization code and datasets</strong>. ðŸ“ŠðŸ’» (Note: Itâ€™s still under construction)</li>
</ol>
<p><img src="https://raw.githubusercontent.com/cafferychen777/ggpicrust2_paper/main/paper_figure/MicrobiomeGallery_preview.jpg"></p>
<ol start="2" style="list-style-type: decimal">
<li>
<a href="https://www.microbiomestat.wiki/" class="external-link">MicrobiomeStat</a>, a <strong>groundbreaking R package designed to revolutionize microbiome analysis</strong>. ðŸ§¬ðŸ”¬ Itâ€™s not just another package; itâ€™s a <strong>game-changer</strong> in the field of microbiome analysis. MicrobiomeStat is <strong>specifically tailored for longitudinal studies</strong>, providing comprehensive solutions for both longitudinal and paired analyses. The package is also adept at handling cross-sectional studies and is <strong>not limited to microbiome data</strong> - it can deal with <strong>multi-omics data</strong> as well. ðŸŽ¯ Our ultimate aim is to make MicrobiomeStat the <strong>go-to R package for longitudinal analysis of multi-omics data</strong>. (Note: Both the package and wiki are still under development)</li>
</ol>
<p>If youâ€™re interested in becoming a <strong>dev user for MicrobiomeStat</strong>, please email <a href="mailto:cafferychen7850@gmail.com" class="email">cafferychen7850@gmail.com</a>. ðŸ“§</p>
<p>Stay tuned for more exciting updates! ðŸŽ‰</p>
<p><img src="https://cdn.staticaly.com/gh/cafferychen777/ggpicrust2_paper/main/paper_figure/WeChatac562b362dadbf21a03414be3a5d06a5.png"></p>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <div class="links">
<h2 data-toc-skip>Links</h2>
<ul class="list-unstyled">
<li><a href="https://cloud.r-project.org/package=ggpicrust2" class="external-link">View on CRAN</a></li>
<li><a href="https://github.com/cafferychen777/ggpicrust2/" class="external-link">Browse source code</a></li>
<li><a href="https://github.com/cafferychen777/ggpicrust2/issues" class="external-link">Report a bug</a></li>
</ul>
</div>

<div class="license">
<h2 data-toc-skip>License</h2>
<ul class="list-unstyled">
<li>
<a href="https://opensource.org/licenses/mit-license.php" class="external-link">MIT</a> + file <a href="LICENSE-text.html">LICENSE</a>
</li>
</ul>
</div>


<div class="citation">
<h2 data-toc-skip>Citation</h2>
<ul class="list-unstyled">
<li><a href="authors.html#citation">Citing ggpicrust2</a></li>
</ul>
</div>

<div class="developers">
<h2 data-toc-skip>Developers</h2>
<ul class="list-unstyled">
<li>Chen Yang <br><small class="roles"> Author, maintainer </small>  </li>
<li>Liangliang Zhang <br><small class="roles"> Author </small>  </li>
</ul>
</div>



  </div>
</div>


      <footer><div class="copyright">
  <p></p>
<p>Developed by Chen Yang, Liangliang Zhang.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
