<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>This function integrates pathway name/description annotations, ten of the most advanced differential abundance (DA) methods, and visualization of DA results. — ggpicrust2 • ggpicrust2</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="This function integrates pathway name/description annotations, ten of the most advanced differential abundance (DA) methods, and visualization of DA results. — ggpicrust2"><meta property="og:description" content="This function integrates pathway name/description annotations, ten of the most advanced differential abundance (DA) methods, and visualization of DA results."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">ggpicrust2</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.7.2</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/using_ggpicrust2.html">ggpicrust2 vigenett</a>
    </li>
  </ul></li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/cafferychen777/ggpicrust2/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>This function integrates pathway name/description annotations, ten of the most advanced differential abundance (DA) methods, and visualization of DA results.</h1>
    <small class="dont-index">Source: <a href="https://github.com/cafferychen777/ggpicrust2/blob/HEAD/R/ggpicrust2.R" class="external-link"><code>R/ggpicrust2.R</code></a></small>
    <div class="hidden name"><code>ggpicrust2.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>This function integrates pathway name/description annotations, ten of the most advanced differential abundance (DA) methods, and visualization of DA results.</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">ggpicrust2</span><span class="op">(</span></span>
<span>  file <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  data <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  <span class="va">metadata</span>,</span>
<span>  <span class="va">group</span>,</span>
<span>  <span class="va">pathway</span>,</span>
<span>  daa_method <span class="op">=</span> <span class="st">"ALDEx2"</span>,</span>
<span>  ko_to_kegg <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  p.adjust <span class="op">=</span> <span class="st">"BH"</span>,</span>
<span>  order <span class="op">=</span> <span class="st">"group"</span>,</span>
<span>  p_values_bar <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  x_lab <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  select <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  reference <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  colors <span class="op">=</span> <span class="cn">NULL</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>file</dt>
<dd><p>A character string representing the file path of the input file containing KO abundance data in picrust2 export format. The input file should have KO identifiers in the first column and sample identifiers in the first row. The remaining cells should contain the abundance values for each KO-sample pair.</p></dd>


<dt>data</dt>
<dd><p>An optional data.frame containing KO abundance data in the same format as the input file. If provided, the function will use this data instead of reading from the file. By default, this parameter is set to NULL.</p></dd>


<dt>metadata</dt>
<dd><p>A tibble, consisting of sample information</p></dd>


<dt>group</dt>
<dd><p>A character, name of the group</p></dd>


<dt>pathway</dt>
<dd><p>A character, consisting of "EC", "KO", "MetaCyc"</p></dd>


<dt>daa_method</dt>
<dd><p>a character specifying the method for differential abundance analysis, default is "ALDEx2", choices are:
- "ALDEx2": ANOVA-Like Differential Expression tool for high throughput sequencing data
- "DESeq2": Differential expression analysis based on the negative binomial distribution using DESeq2
- "edgeR": Exact test for differences between two groups of negative-binomially distributed counts using edgeR
- "limma voom": Limma-voom framework for the analysis of RNA-seq data
- "metagenomeSeq": Fit logistic regression models to test for differential abundance between groups using metagenomeSeq
- "LinDA": Linear models for differential abundance analysis of microbiome compositional data
- "Maaslin2": Multivariate Association with Linear Models (MaAsLin2) for differential abundance analysis</p></dd>


<dt>ko_to_kegg</dt>
<dd><p>A character to control the conversion of KO abundance to KEGG abundance</p></dd>


<dt>p.adjust</dt>
<dd><p>a character specifying the method for p-value adjustment, default is "BH", choices are:
- "BH": Benjamini-Hochberg correction
- "holm": Holm's correction
- "bonferroni": Bonferroni correction
- "hochberg": Hochberg's correction
- "fdr": False discovery rate correction
- "none": No p-value adjustment.</p></dd>


<dt>order</dt>
<dd><p>A character to control the order of the main plot rows</p></dd>


<dt>p_values_bar</dt>
<dd><p>A character to control if the main plot has the p_values bar</p></dd>


<dt>x_lab</dt>
<dd><p>A character to control the x-axis label name, you can choose from "feature","pathway_name" and "description"</p></dd>


<dt>select</dt>
<dd><p>A vector consisting of pathway names to be selected</p></dd>


<dt>reference</dt>
<dd><p>A character, a reference group level for several DA methods</p></dd>


<dt>colors</dt>
<dd><p>A vector consisting of colors number</p></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    

<p>daa.results.df, a dataframe of DA results
A list of sub-lists, each containing a ggplot2 plot (`plot`) and a dataframe of differential abundance results (`results`) for a specific DA method.
Each plot visualizes the differential abundance results of a specific DA method, and the corresponding dataframe contains the results used to create the plot.</p>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="kw">if</span> <span class="op">(</span><span class="cn">FALSE</span><span class="op">)</span> <span class="op">{</span></span></span>
<span class="r-in"><span><span class="co"># Load necessary data: abundance data and metadata</span></span></span>
<span class="r-in"><span><span class="va">abundance_file</span> <span class="op">&lt;-</span> <span class="st">"path/to/your/abundance_file.tsv"</span></span></span>
<span class="r-in"><span><span class="va">metadata</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.csv</a></span><span class="op">(</span><span class="st">"path/to/your/metadata.csv"</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Run ggpicrust2 with input file path</span></span></span>
<span class="r-in"><span><span class="va">results_file_input</span> <span class="op">&lt;-</span> <span class="fu">ggpicrust2</span><span class="op">(</span>file <span class="op">=</span> <span class="va">abundance_file</span>,</span></span>
<span class="r-in"><span>                                 metadata <span class="op">=</span> <span class="va">metadata</span>,</span></span>
<span class="r-in"><span>                                 group <span class="op">=</span> <span class="st">"your_group_column"</span>,</span></span>
<span class="r-in"><span>                                 pathway <span class="op">=</span> <span class="st">"KO"</span>,</span></span>
<span class="r-in"><span>                                 daa_method <span class="op">=</span> <span class="st">"LinDA"</span>,</span></span>
<span class="r-in"><span>                                 ko_to_kegg <span class="op">=</span> <span class="st">"TRUE"</span>,</span></span>
<span class="r-in"><span>                                 order <span class="op">=</span> <span class="st">"pathway_class"</span>,</span></span>
<span class="r-in"><span>                                 p_values_bar <span class="op">=</span> <span class="cn">TRUE</span>,</span></span>
<span class="r-in"><span>                                 x_lab <span class="op">=</span> <span class="st">"pathway_name"</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Run ggpicrust2 with imported data.frame</span></span></span>
<span class="r-in"><span><span class="va">abundance_data</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://readr.tidyverse.org/reference/read_delim.html" class="external-link">read_delim</a></span><span class="op">(</span><span class="va">abundance_file</span>, delim<span class="op">=</span><span class="st">"\t"</span>, col_names<span class="op">=</span><span class="cn">TRUE</span>, trim_ws<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Run ggpicrust2 with input data</span></span></span>
<span class="r-in"><span><span class="va">results_data_input</span> <span class="op">&lt;-</span> <span class="fu">ggpicrust2</span><span class="op">(</span>data <span class="op">=</span> <span class="va">abundance_data</span>,</span></span>
<span class="r-in"><span>                                 metadata <span class="op">=</span> <span class="va">metadata</span>,</span></span>
<span class="r-in"><span>                                 group <span class="op">=</span> <span class="st">"your_group_column"</span>,</span></span>
<span class="r-in"><span>                                 pathway <span class="op">=</span> <span class="st">"KO"</span>,</span></span>
<span class="r-in"><span>                                 daa_method <span class="op">=</span> <span class="st">"LinDA"</span>,</span></span>
<span class="r-in"><span>                                 ko_to_kegg <span class="op">=</span> <span class="st">"TRUE"</span>,</span></span>
<span class="r-in"><span>                                 order <span class="op">=</span> <span class="st">"pathway_class"</span>,</span></span>
<span class="r-in"><span>                                 p_values_bar <span class="op">=</span> <span class="cn">TRUE</span>,</span></span>
<span class="r-in"><span>                                 x_lab <span class="op">=</span> <span class="st">"pathway_name"</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Access the plot and results dataframe for the first DA method</span></span></span>
<span class="r-in"><span><span class="va">example_plot</span> <span class="op">&lt;-</span> <span class="va">results_file_input</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">$</span><span class="va">plot</span></span></span>
<span class="r-in"><span><span class="va">example_results</span> <span class="op">&lt;-</span> <span class="va">results_file_input</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">$</span><span class="va">results</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Use the example data in ggpicrust2 package</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">ko_abundance</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">metadata</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">results_file_input</span> <span class="op">&lt;-</span> <span class="fu">ggpicrust2</span><span class="op">(</span>data <span class="op">=</span> <span class="va">ko_abundance</span>,</span></span>
<span class="r-in"><span>                                 metadata <span class="op">=</span> <span class="va">metadata</span>,</span></span>
<span class="r-in"><span>                                 group <span class="op">=</span> <span class="st">"Environment"</span>,</span></span>
<span class="r-in"><span>                                 pathway <span class="op">=</span> <span class="st">"KO"</span>,</span></span>
<span class="r-in"><span>                                 daa_method <span class="op">=</span> <span class="st">"LinDA"</span>,</span></span>
<span class="r-in"><span>                                 ko_to_kegg <span class="op">=</span> <span class="cn">TRUE</span>,</span></span>
<span class="r-in"><span>                                 order <span class="op">=</span> <span class="st">"pathway_class"</span>,</span></span>
<span class="r-in"><span>                                 p_values_bar <span class="op">=</span> <span class="cn">TRUE</span>,</span></span>
<span class="r-in"><span>                                 x_lab <span class="op">=</span> <span class="st">"pathway_name"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="co"># Analyze the EC or MetaCyc pathway</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">metacyc_abundance</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">results_file_input</span> <span class="op">&lt;-</span> <span class="fu">ggpicrust2</span><span class="op">(</span>data <span class="op">=</span> <span class="va">metacyc_abundance</span>,</span></span>
<span class="r-in"><span>                                 metadata <span class="op">=</span> <span class="va">metadata</span>,</span></span>
<span class="r-in"><span>                                 group <span class="op">=</span> <span class="st">"Environment"</span>,</span></span>
<span class="r-in"><span>                                 pathway <span class="op">=</span> <span class="st">"MetaCyc"</span>,</span></span>
<span class="r-in"><span>                                 daa_method <span class="op">=</span> <span class="st">"LinDA"</span>,</span></span>
<span class="r-in"><span>                                 ko_to_kegg <span class="op">=</span> <span class="cn">FALSE</span>,</span></span>
<span class="r-in"><span>                                 order <span class="op">=</span> <span class="st">"group"</span>,</span></span>
<span class="r-in"><span>                                 p_values_bar <span class="op">=</span> <span class="cn">TRUE</span>,</span></span>
<span class="r-in"><span>                                 x_lab <span class="op">=</span> <span class="st">"description"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="op">}</span></span></span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Chen Yang, Liangliang Zhang.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer></div>

  


  

  </body></html>

