<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Predictional functional patwhay differential abundance (DA) — pathway_daa • ggpicrust2</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Predictional functional patwhay differential abundance (DA) — pathway_daa"><meta property="og:description" content="Predictional functional patwhay differential abundance (DA)"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">ggpicrust2</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.7.2</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/using_ggpicrust2.html">ggpicrust2 vigenett</a>
    </li>
  </ul></li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/cafferychen777/ggpicrust2/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Predictional functional patwhay differential abundance (DA)</h1>
    <small class="dont-index">Source: <a href="https://github.com/cafferychen777/ggpicrust2/blob/HEAD/R/pathway_daa.R" class="external-link"><code>R/pathway_daa.R</code></a></small>
    <div class="hidden name"><code>pathway_daa.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Predictional functional patwhay differential abundance (DA)</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">pathway_daa</span><span class="op">(</span></span>
<span>  <span class="va">abundance</span>,</span>
<span>  <span class="va">metadata</span>,</span>
<span>  <span class="va">group</span>,</span>
<span>  daa_method <span class="op">=</span> <span class="st">"ALDEx2"</span>,</span>
<span>  select <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  p.adjust <span class="op">=</span> <span class="st">"BH"</span>,</span>
<span>  reference <span class="op">=</span> <span class="cn">NULL</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>abundance</dt>
<dd><p>a data frame containing predicted functional pathway abundance, with pathways/features as rows and samples as columns. The column names of abundance should match the sample names in metadata. Pathway abundance values should be counts</p></dd>


<dt>metadata</dt>
<dd><p>a tibble containing samples information</p></dd>


<dt>group</dt>
<dd><p>a character specifying the group name for differential abundance analysis</p></dd>


<dt>daa_method</dt>
<dd><p>a character specifying the method for differential abundance analysis, choices are:
- "ALDEx2": ANOVA-Like Differential Expression tool for high throughput sequencing data
- "DESeq2": Differential expression analysis based on the negative binomial distribution using DESeq2
- "edgeR": Exact test for differences between two groups of negative-binomially distributed counts using edgeR
- "limma voom": Limma-voom framework for the analysis of RNA-seq data
- "metagenomeSeq": Fit logistic regression models to test for differential abundance between groups using metagenomeSeq
- "LinDA": Linear models for differential abundance analysis of microbiome compositional data
- "Maaslin2": Multivariate Association with Linear Models (MaAsLin2) for differential abundance analysis
- "Lefser": Linear discriminant analysis (LDA) effect size algorithm for high-dimensional microbiome data</p></dd>


<dt>select</dt>
<dd><p>a vector containing sample names for analysis, if NULL all samples are included. This parameter can be used to specify which samples are included in the differential abundance analysis. Default is NULL.</p></dd>


<dt>p.adjust</dt>
<dd><p>a character specifying the method for p-value adjustment, choices are:
- "BH": Benjamini-Hochberg correction
- "holm": Holm's correction
- "bonferroni": Bonferroni correction
- "hochberg": Hochberg's correction
- "fdr": False discovery rate correction
- "none": No p-value adjustment.</p></dd>


<dt>reference</dt>
<dd><p>a character specifying the reference group level, required for several differential abundance analysis methods such as LinDA, limme voom and Maaslin2. This parameter is used to specify the reference group when there are more than two groups. Default is NULL.</p></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    

<p>a data frame containing the differential abundance analysis results.</p>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="co"># \donttest{</span></span></span>
<span class="r-in"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/cafferychen777/ggpicrust2" class="external-link">ggpicrust2</a></span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">MicrobiomeStat</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Registered S3 method overwritten by 'rmutil':</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   method         from</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   print.response httr</span>
<span class="r-in"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://tibble.tidyverse.org/" class="external-link">tibble</a></span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://magrittr.tidyverse.org" class="external-link">magrittr</a></span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">abundance</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>sample1 <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">10</span>, <span class="fl">20</span>, <span class="fl">30</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>sample2 <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">20</span>, <span class="fl">30</span>, <span class="fl">40</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>sample3 <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">30</span>, <span class="fl">40</span>, <span class="fl">50</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>row.names <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"pathway1"</span>, <span class="st">"pathway2"</span>, <span class="st">"pathway3"</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="va">metadata</span> <span class="op">&lt;-</span> <span class="fu">tibble</span><span class="fu">::</span><span class="fu"><a href="https://tibble.tidyverse.org/reference/tibble.html" class="external-link">tibble</a></span><span class="op">(</span>sample <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"sample"</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">3</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>group <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"control"</span>, <span class="st">"control"</span>, <span class="st">"treatment"</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">#Run pathway_daa function</span></span></span>
<span class="r-in"><span><span class="va">result</span> <span class="op">&lt;-</span> <span class="fu">pathway_daa</span><span class="op">(</span>abundance <span class="op">=</span> <span class="va">abundance</span>, metadata <span class="op">=</span> <span class="va">metadata</span>, group <span class="op">=</span> <span class="st">"group"</span>,</span></span>
<span class="r-in"><span>daa_method <span class="op">=</span> <span class="st">"LinDA"</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Sample names extracted.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Identifying matching columns in metadata...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Matching columns identified: sample . This is important for ensuring data consistency.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Using all columns in abundance.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Converting abundance to a matrix...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Reordering metadata...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Converting metadata to a matrix and data frame...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Extracting group information...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Running LinDA analysis...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Performing LinDA analysis...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 0  features are filtered!</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> The filtered data has  3  samples and  3  features will be tested!</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Fit linear models ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Completed.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Processing LinDA results...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> LinDA analysis is complete.</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">metacyc_abundance</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">metadata</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">daa_results_df</span> <span class="op">&lt;-</span> <span class="fu">pathway_daa</span><span class="op">(</span><span class="va">metacyc_abundance</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://tibble.tidyverse.org/reference/rownames.html" class="external-link">column_to_rownames</a></span><span class="op">(</span><span class="st">"pathway"</span><span class="op">)</span>, <span class="va">metadata</span>, <span class="st">"Environment"</span>, daa_method <span class="op">=</span> <span class="st">"Lefser"</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Sample names extracted.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Identifying matching columns in metadata...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Matching columns identified: NA . This is important for ensuring data consistency.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Using all columns in abundance.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Converting abundance to a matrix...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Reordering metadata...</span>
<span class="r-err co"><span class="r-pr">#&gt;</span> <span class="error">Error in metadata[, matching_columns]:</span> Can't subset columns with `matching_columns`.</span>
<span class="r-err co"><span class="r-pr">#&gt;</span> <span style="color: #BB0000;">✖</span> Subscript `matching_columns` can't contain missing values.</span>
<span class="r-err co"><span class="r-pr">#&gt;</span> <span style="color: #BB0000;">✖</span> It has a missing value at location 1.</span>
<span class="r-in"><span><span class="co"># }</span></span></span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Chen Yang, Liangliang Zhang.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer></div>

  


  

  </body></html>

